src/App.js

```diff
@@ -71,87 +65,18 @@ function App() {
       >
         <Layout>
           <GlobalStyles />
-          <Router>
-            <div className="App">
-              <Switch>
-                {!token && (
-                  <Route path={['/list', '/additems', '/shareyourtoken']}>
-                    <Redirect to="/" />
-                  </Route>
-                )}
-                <Route path="/list">
-                  <ItemList
-                    list={list}
-                    loading={loading}
-                    error={error}
-                    userToken={token}
-                  />
-                </Route>
-                <Route path="/shareyourtoken">
-                  <ShareYourToken />
-                </Route>
-                <Route path="/additems">
-                  <AddItems list={list} userToken={token} />
-                </Route>
-                <Route exact path="/">
-                  {token ? (
-                    <Redirect to="/list" />
-                  ) : (
-                    <Welcome updateToken={updateToken} />
-                  )}
-                </Route>
-              </Switch>
-              {token && (
-                <nav
-                  style={{
-                    position: 'fixed',
-                    bottom: '0',
-                    width: '100%',
-                    backgroundColor: '#fff',
-                  }}
-                >
-                  <ul
-                    style={{
-                      display: 'flex',
-                      justifyContent: 'space-evenly',
-                      paddingLeft: '0',
-                    }}
-                  >
-                    <li>
-                      <NavLink
-                        to="/list"
-                        activeStyle={{
-                          fontWeight: 'bold',
-                        }}
-                      >
-                        Grocery List
-                      </NavLink>
-                    </li>
-                    <li>
-                      <NavLink
-                        to="/shareyourtoken"
-                        activeStyle={{
-                          fontWeight: 'bold',
-                        }}
-                      >
-                        Share Your Token
-                      </NavLink>
-                    </li>
-                    <li>
-                      <NavLink
-                        to="/additems"
-                        activeStyle={{
-                          fontWeight: 'bold',
-                        }}
-                      >
-                        Add Groceries
-                      </NavLink>
-                    </li>
-                  </ul>
-                </nav>
-              )}
-            </div>
-          </Router>
+          <div className="App">
+            <h1>Shopping app</h1>
+            <RouterComponent
```

Comment:
so annoying that you had to refactor the whole router! great job getting it to work though!